tools-local/tasks/net45/ProcessSharedFrameworkDeps.net45.cs

```diff
@@ -0,0 +1,25 @@
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Common;
+using NuGet.ProjectModel;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class ProcessSharedFrameworkDeps
+    {
+        partial void EnsureInitialized(string buildToolsTaskDir)
+        {
+            // ensure the build tools AssemblyResolver is enabled, so we get the correct assembly unification
+            // even if the build tools assembly hasn't been loaded yet.
+            string buildTasksPath = Path.Combine(buildToolsTaskDir, "Microsoft.DotNet.Build.Tasks.dll");
+            Assembly buildTasksAssembly = Assembly.Load(AssemblyName.GetAssemblyName(buildTasksPath));
+            Type assemblyResolver = buildTasksAssembly.GetType("Microsoft.DotNet.Build.Common.Desktop.AssemblyResolver");
```

Comment:
I thought of copying the file into core-setup, but I wasn't crazy about it for obvious reasons (duplication of code, duplication of logic running during our build - since there would be 2 AssemblyResolvers that were "enabled").

I think the approach I would have preferred is for this to be exposed in a NuGet package that this tasks library can reference...  But I think our plan is to pull these tasks into build-tools anyway, right?  When that happens, this can just call AssemblyResolver.Enable directly.